src/scripts/manifest.js

```diff
@@ -0,0 +1,190 @@
+
+                    let output = "Issues found:\n";
+
+                    for (let i = 0; i < manifestValidationResult['type'].length; i++) {
+                        if (manifestValidationResult['type'][i] === 'missingFields') {
+                            let missingFields = manifestValidationResult['missingFields'].join(', ');
+                            output += `- Missing fields: ${missingFields}\n`;
+                        }
+                        if (manifestValidationResult['type'][i] === 'invalidValueTypes') {
+                            let invalidValueTypes = manifestValidationResult['invalidValueTypes'].join(', ');
+                            output += `- Invalid value types for: ${invalidValueTypes}\n`;
+                        }
+                    }
+                    
+                    fileInfoDiv.innerText = output;
+                        
+                }
+            } catch(error){
+                console.log("Error parsing manifest JSON: ", error);
+                document.getElementById('manifestFileInfo').textContent = 'Error parsing manifest.json file';
+            }
+        }
+        reader.readAsText(manifestFile);
+    }
+});
+
+// Display the paths for the ruleset files, as defined in the manifest
+function displayRulesetFilePaths(manifest){
+    const fileInfo = document.getElementById('manifestFileInfo');
+    let output = 'Ruleset Files:\n';    
+    if(manifest.declarative_net_request.rule_resources){
+        manifest.declarative_net_request.rule_resources.forEach(ruleset => {
+            output += `- ${ruleset.id}: ${ruleset.path}, Enabled: ${ruleset.enabled}\n`;
+            rulesetFilePaths.push({rulesetFilePath: ruleset.path, rulesetId: ruleset.id});
+        });
+        fileInfo.innerText = output;     
+    }       
+}
+
+// Returns true if the manifest is valid, else returns an object with the errors
+function isValidManifest(manifest) {
+    let syntaxError = {};
+    syntaxError['type'] = [];
+
+    // Check for declarativeNetRequest and declarativeNetRequestWithHostAccess permissions
+    if(!manifest.permissions.includes('declarativeNetRequest') && !manifest.permissions.includes('declarativeNetRequestWithHostAccess')){
+        syntaxError.isError = true;
+        syntaxError['type'].push('missingPermissions');
+    }
+
+    // Check for required fields
+    const requiredFieldsAndTypes = { // "description" and "icon" required for uploading to web store
+        "name": "string",
+        "version": "string",
+        "manifest_version": "number",
+    }
+    for (let field of Object.keys(requiredFieldsAndTypes)) {
+        if (!manifest.hasOwnProperty(field)) {
+            syntaxError.isError = true;
+            syntaxError['type'].push('missingFields');
+            syntaxError['missingFields'] = [];
+            syntaxError['missingFields'].push(field);
+        } else if(manifest.hasOwnProperty(field) && (typeof manifest[field] !== requiredFieldsAndTypes[field])){
```

Comment:
```suggestion
        } else if(typeof manifest[field] !== requiredFieldsAndTypes[field]){
```

I think we can do this, since we already checked this above?

Reply:
Resolving so I can merge.